docs/raml/traits.yml

```diff
@@ -1,6 +1,49 @@
 #%RAML 0.8
 
 
+- jobQueryer:
+    queryParameters:
+      categories:
+        type: string
+        description: the categories that jobs must have, expressed as a singly url encoded string conforming to categoriesActual.json. 
+        example: |
+          %5B%7B%22category%22%3A%20100001999%2C%20%22values%22%3A%20%5B100115543%2C100115516%5D%7D%5D
+        required: false
+      keywords:
+        type: string
+        description: keywords that the jobs must have
+        example: java architect
+        required: false
+      minPublishDate:
+        type: date
+        description: a date, in rfc7095 format, used to filter jobs (must not be combined with any other date filters). Only jobs with a date greater than this will be returned.
+        example: |
+          2015-11-05T13:15:30Z 
+        required: false
+      minInternalPublishDate:
+        type: date
+        description: a date, in rfc7095 format, used to filter jobs (must not be combined with any other date filters). Only jobs with a date greater than this will be returned.
+        example: |
+          2015-11-05T13:15:30Z 
+        required: false
+      minExternalPublishDate:
+        type: date
+        description: a date, in rfc7095 format, used to filter jobs (must not be combined with any other date filters). Only jobs with a date greater than this will be returned.
+        example: |
+          2015-11-05T13:15:30Z 
+        required: false
+      minID:
+        type: number
+        required: false
+        example: 1001
+        description: if present, only include applications with id > this value 
```

Comment:
`>=`? I usually expect a lower bound to be inclusive and upper bound exclusive

Reply:
True in Java but for API pagination you want it the other way around.

e.g.
...first fetch returns A, B and C
...then you pass "C" into the next fetch
...and get back D, E and F

Reply:
Maybe its the parameter name throwing me off. Could something like `after` be used? Or the response body could contain the first id of the next page e.g. https://developer.atlassian.com/confdev/confluence-server-rest-api/pagination-in-the-rest-api

Reply:
I'm open to changing the param name, since "min" is misleading, but I don't know what to. after... doesn't parse well.

But as for the semantics, I don't want to go down the road suggested since:

a) its unnecessary to pass back the first ID of the next page, since you can do it just as well by using a > in the where clause in the underlying database.

b) doing so causes unnecessary extra network traffic

c) its also potentially less accurate. A new first item may arrive on the next page, and you'll never see it if you already think you know the first item of the next page. Doing it as per the current system means that if something is inserted into the set ahead of you, you'll always see it.

Reply:
`IDgreaterThan`?

Reply:
verbose.. how about "gt" (greater than)

gtID
gtPublishDate

or...

idgt
publisDategt

I think, looking at both of these, that the former is more readable, and also suitably emphasises the behaviour.

Reply:
I'm happy with that